lib/sisimai/bite/email/sendgrid.rb

```diff
@@ -0,0 +1,223 @@
+                # Final-Recipient: RFC822; userunknown@example.jp
+                if v['recipient']
+                  # There are multiple recipient addresses in the message body.
+                  dscontents << Sisimai::Bite.DELIVERYSTATUS
+                  v = dscontents[-1]
+                end
+                v['recipient'] = cv[1]
+                recipients += 1
+
+              elsif cv = e.match(/\A[Aa]ction:[ ]*(.+)\z/)
+                # Action: failed
+                v['action'] = cv[1].downcase
+
+              elsif cv = e.match(/\A[Ss]tatus:[ ]*(\d[.]\d+[.]\d+)/)
+                # Status: 5.1.1
+                # Status:5.2.0
+                # Status: 5.1.0 (permanent failure)
+                v['status'] = cv[1]
+
+              else
+                if cv = e.match(/\A[Dd]iagnostic-[Cc]ode:[ ]*(.+)\z/)
+                  # Diagnostic-Code: 550 5.1.1 <userunknown@example.jp>... User Unknown
+                  v['diagnosis'] = cv[1]
+
+                elsif p =~ /\A[Dd]iagnostic-[Cc]ode:[ ]*/ && cv = e.match(/\A[ \t]+(.+)\z/)
+                  # Continued line of the value of Diagnostic-Code header
+                  v['diagnosis'] ||= ''
+                  v['diagnosis']  += ' ' + cv[1]
+                  havepassed[-1] = 'Diagnostic-Code: ' + e
+                end
+              end
+            else
+              # This is an automatically generated message from SendGrid.
+              #
+              # I'm sorry to have to tell you that your message was not able to be
+              # delivered to one of its intended recipients.
+              #
+              # If you require assistance with this, please contact SendGrid support.
+              #
+              # shironekochan:000000:<kijitora@example.jp> : 192.0.2.250 : mx.example.jp:[192.0.2.153] :
+              #   550 5.1.1 <userunknown@cubicroot.jp>... User Unknown  in RCPT TO
+              #
+              # ------------=_1351676802-30315-116783
+              # Content-Type: message/delivery-status
+              # Content-Disposition: inline
+              # Content-Transfer-Encoding: 7bit
+              # Content-Description: Delivery Report
+              #
+              # X-SendGrid-QueueID: 959479146
+              # X-SendGrid-Sender: <bounces+61689-10be-kijitora=example.jp@sendgrid.info>
+              # Arrival-Date: 2012-12-31 23-59-59
+              if cv = e.match(/.+ in (?:End of )?([A-Z]{4}).*\z/)
+                # in RCPT TO, in MAIL FROM, end of DATA
+                commandtxt = cv[1]
+
+              elsif cv = e.match(/\A[Aa]rrival-[Dd]ate:[ ]*(.+)\z/)
+                # Arrival-Date: Wed, 29 Apr 2009 16:03:18 +0900
+                next if connheader['date'].size > 0
+                arrivaldate = cv[1]
+
+                if cv = e.match(/\A[Aa]rrival-[Dd]ate: (\d{4})[-](\d{2})[-](\d{2}) (\d{2})[-](\d{2})[-](\d{2})\z/)
+                  # Arrival-Date: 2011-08-12 01-05-05
```

Comment:
__[rubocop]__
* Style/IfInsideElse: Convert `if` nested inside `else` to `elsif`.

<!-- Comment by [SideCI](https://sideci.com) -->